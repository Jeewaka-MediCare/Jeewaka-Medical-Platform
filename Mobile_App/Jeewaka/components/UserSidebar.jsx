import React, { useEffect, useRef } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Modal,
  Animated,
  Dimensions,
  TouchableWithoutFeedback,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import useAuthStore from '../store/authStore';

const UserDropdown = ({ visible, onClose, onLogin, onLogout }) => {
  const { user, userRole } = useAuthStore();
  const scaleAnim = useRef(new Animated.Value(0)).current;
  const opacityAnim = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (visible) {
      Animated.parallel([
        Animated.spring(scaleAnim, {
          toValue: 1,
          useNativeDriver: true,
          tension: 120,
          friction: 8,
        }),
        Animated.timing(opacityAnim, {
          toValue: 1,
          duration: 200,
          useNativeDriver: true,
        })
      ]).start();
    } else {
      Animated.parallel([
        Animated.timing(scaleAnim, {
          toValue: 0,
          duration: 150,
          useNativeDriver: true,
        }),
        Animated.timing(opacityAnim, {
          toValue: 0,
          duration: 150,
          useNativeDriver: true,
        })
      ]).start();
    }
  }, [visible]);

  const getDisplayName = () => {
    const nameOptions = [
      user?.name,
      user?.fullName,
      user?.firstName,
      user?.email
    ];
    
    for (const nameOption of nameOptions) {
      if (nameOption && typeof nameOption === 'string' && nameOption.trim()) {
        return nameOption.trim();
      }
    }
    
    return userRole === 'doctor' ? 'Doctor' : 'Patient';
  };

  const getAvatarInitial = () => {
    const nameOptions = [
      user?.name,
      user?.fullName, 
      user?.firstName,
      user?.email?.charAt(0)
    ];
    
    for (const nameOption of nameOptions) {
      if (nameOption && typeof nameOption === 'string' && nameOption.trim()) {
        return nameOption.charAt(0).toUpperCase();
      }
    }
    
    return userRole === 'doctor' ? 'D' : 'P';
  };

  const getRoleDisplay = () => {
    return userRole === 'doctor' ? 'Doctor' : 'Patient';
  };

  return (
    <Modal
      animationType="none"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <TouchableWithoutFeedback onPress={onClose}>
        <View style={styles.overlay}>
          <Animated.View 
            style={[
              styles.dropdown,
              {
                transform: [{ scale: scaleAnim }],
                opacity: opacityAnim,
              }
            ]}
          >
            {user ? (
              // Logged in user content
              <>
                <View style={styles.userInfo}>
                  <View style={styles.avatar}>
                    <Text style={styles.avatarText}>{getAvatarInitial()}</Text>
                  </View>
                  <View style={styles.nameContainer}>
                    <Text style={styles.userName}>{getDisplayName()}</Text>
                    <Text style={styles.userRole}>{getRoleDisplay()}</Text>
                  </View>
                </View>
                
                <View style={styles.divider} />
                
                <TouchableOpacity 
                  style={styles.menuItem} 
                  onPress={onLogout}
                >
                  <Ionicons name="log-out-outline" size={18} color="#EF4444" />
                  <Text style={styles.logoutText}>Logout</Text>
                </TouchableOpacity>
              </>
            ) : (
              // Not logged in content
              <>
                <View style={styles.guestInfo}>
                  <Ionicons name="person-circle-outline" size={24} color="#64748B" />
                  <Text style={styles.guestText}>Not logged in</Text>
                </View>
                
                <View style={styles.divider} />
                
                <TouchableOpacity 
                  style={styles.menuItem} 
                  onPress={onLogin}
                >
                  <Ionicons name="log-in-outline" size={18} color="#008080" />
                  <Text style={styles.loginText}>Login</Text>
                </TouchableOpacity>
              </>
            )}
          </Animated.View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'transparent',
    justifyContent: 'flex-start',
    alignItems: 'flex-end',
    paddingTop: 60, // Account for header height
    paddingRight: 16,
  },
  dropdown: {
    backgroundColor: 'white',
    borderRadius: 12,
    padding: 16,
    minWidth: 200,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.25,
    shadowRadius: 8,
    elevation: 8,
  },
  userInfo: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 8,
  },
  avatar: {
    width: 36,
    height: 36,
    borderRadius: 18,
    backgroundColor: '#008080',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 12,
  },
  avatarText: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
  },
  nameContainer: {
    flex: 1,
  },
  userName: {
    fontSize: 16,
    fontWeight: '600',
    color: '#1F2937',
    marginBottom: 2,
  },
  userRole: {
    fontSize: 12,
    color: '#64748B',
    textTransform: 'capitalize',
  },
  guestInfo: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 8,
  },
  guestText: {
    fontSize: 14,
    color: '#64748B',
    marginLeft: 8,
  },
  divider: {
    height: 1,
    backgroundColor: '#E2E8F0',
    marginVertical: 12,
  },
  menuItem: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 8,
  },
  logoutText: {
    fontSize: 14,
    color: '#EF4444',
    marginLeft: 8,
    fontWeight: '500',
  },
  loginText: {
    fontSize: 14,
    color: '#008080',
    marginLeft: 8,
    fontWeight: '500',
  },
});

export default UserDropdown;