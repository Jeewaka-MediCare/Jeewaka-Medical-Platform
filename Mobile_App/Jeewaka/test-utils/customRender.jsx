import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

// Custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const Wrapper = ({ children }) => {
    return (
      <NavigationContainer>
        {children}
      </NavigationContainer>
    );
  };

  return render(ui, { wrapper: Wrapper, ...options });
};

// Re-export everything
export * from '@testing-library/react-native';

// Override render method
export { renderWithProviders as render };