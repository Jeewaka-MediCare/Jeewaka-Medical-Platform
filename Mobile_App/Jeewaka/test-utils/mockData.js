// Mock data for testing

export const mockDoctor = {
  _id: '507f1f77bcf86cd799439011',
  uid: 'doctor-uid-123',
  name: 'Dr. John Smith',
  email: 'john.smith@example.com',
  specialization: 'Cardiology',
  subSpecializations: ['Interventional Cardiology', 'Heart Surgery'],
  yearsOfExperience: 15,
  qualifications: ['MBBS', 'MD Cardiology', 'FRCPC'],
  hospitalAffiliation: 'General Hospital',
  consultationFee: 150,
  avgRating: 4.5,
  totalRatings: 120,
  totalReviews: 120,
  profilePicture: 'https://example.com/doctor.jpg',
  bio: 'Experienced cardiologist with 15 years of practice.',
  isAvailable: true,
  gender: 'Male',
  languages: ['English', 'Spanish'],
  phoneNumber: '+1234567890',
  address: '123 Medical St, City, State 12345',
  dateOfBirth: '1975-05-15',
  licenseNumber: 'MD12345',
  isVerified: true,
  createdAt: '2023-01-01T00:00:00.000Z',
  updatedAt: '2024-01-01T00:00:00.000Z',
};

export const mockPatient = {
  _id: '507f1f77bcf86cd799439012',
  uid: 'patient-uid-123',
  name: 'Jane Doe',
  email: 'jane.doe@example.com',
  phoneNumber: '+1234567891',
  dateOfBirth: '1990-03-20',
  gender: 'Female',
  address: '456 Patient Ave, City, State 12345',
  emergencyContact: {
    name: 'John Doe',
    relationship: 'Spouse',
    phoneNumber: '+1234567892',
  },
  medicalHistory: ['Diabetes', 'Hypertension'],
  allergies: ['Penicillin'],
  currentMedications: ['Metformin', 'Lisinopril'],
  insuranceInfo: {
    provider: 'Health Insurance Co',
    policyNumber: 'POL123456',
  },
  createdAt: '2023-01-01T00:00:00.000Z',
  updatedAt: '2024-01-01T00:00:00.000Z',
};

export const mockAppointment = {
  _id: '507f1f77bcf86cd799439013',
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  date: '2024-02-15',
  startTime: '10:00 AM',
  endTime: '10:30 AM', 
  duration: 30,
  type: 'consultation',
  status: 'confirmed',
  consultationFee: 150,
  paymentStatus: 'paid',
  notes: 'Regular checkup',
  sessionId: 'session-123',
  doctorName: mockDoctor.name,
  doctorSpecialization: mockDoctor.specialization,
  patientName: mockPatient.name,
  isOngoing: false,
  createdAt: '2024-01-01T00:00:00.000Z',
  updatedAt: '2024-01-01T00:00:00.000Z',
};

export const mockPayment = {
  _id: '507f1f77bcf86cd799439014',
  appointmentId: mockAppointment._id,
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  amount: 150,
  currency: 'LKR',
  paymentMethod: 'stripe',
  stripePaymentIntentId: 'pi_test_123',
  status: 'completed',
  doctorName: mockDoctor.name,
  doctorSpecialization: mockDoctor.specialization,
  date: '2024-01-01',
  appointmentDate: '2024-02-15',
  type: 'consultation',
  paidAt: '2024-01-01T12:00:00.000Z',
  createdAt: '2024-01-01T12:00:00.000Z',
};

export const mockReview = {
  _id: '507f1f77bcf86cd799439015',
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  appointmentId: mockAppointment._id,
  rating: 5,
  comment: 'Excellent doctor, very professional and caring.',
  createdAt: '2024-01-02T00:00:00.000Z',
};

export const mockDoctorsList = [
  mockDoctor,
  {
    ...mockDoctor,
    _id: '507f1f77bcf86cd799439016',
    uid: 'doctor-uid-456',
    name: 'Dr. Sarah Johnson',
    email: 'sarah.johnson@example.com',
    specialization: 'Dermatology',
    subSpecializations: ['Cosmetic Dermatology', 'Pediatric Dermatology'],
    consultationFee: 120,
    avgRating: 4.8,
    totalRatings: 95,
    totalReviews: 95, // Add both fields for compatibility
  },
  {
    ...mockDoctor,
    _id: '507f1f77bcf86cd799439017',
    uid: 'doctor-uid-789',
    name: 'Dr. Michael Brown',
    email: 'michael.brown@example.com',
    specialization: 'Pediatrics',
    subSpecializations: ['Neonatology', 'Pediatric Cardiology'],
    consultationFee: 100,
    avgRating: 4.6,
    totalRatings: 87,
    totalReviews: 87, // Add both fields for compatibility
  },
];

export const mockApiResponse = {
  success: true,
  data: mockDoctorsList,
  message: 'Doctors fetched successfully',
};

export const mockApiError = {
  success: false,
  error: 'Network error',
  message: 'Failed to fetch data',
};

export const mockAuthUser = {
  uid: 'user-uid-123',
  email: 'test@example.com',
  displayName: 'Test User',
  emailVerified: true,
  getIdToken: jest.fn(() => Promise.resolve('mock-token')),
  getIdTokenResult: jest.fn(() => Promise.resolve({
    claims: { role: 'patient' }
  })),
};

export const mockMedicalRecord = {
  _id: '507f1f77bcf86cd799439018',
  patientId: mockPatient._id,
  doctorId: mockDoctor._id,
  appointmentId: mockAppointment._id,
  title: 'Cardiology Consultation',
  diagnosis: 'Hypertension',
  treatment: 'Prescribed medication and lifestyle changes',
  medications: ['Lisinopril 10mg', 'Amlodipine 5mg'],
  notes: 'Patient shows good response to treatment',
  labResults: 'Blood pressure: 130/80 mmHg',
  attachments: ['test-result.pdf'],
  date: '2024-02-15',
  createdAt: '2024-02-15T10:30:00.000Z',
  updatedAt: '2024-02-15T10:30:00.000Z',
};

export const mockVideoCallSession = {
  sessionId: 'session-123',
  participantId: 'participant-456',
  meetingId: 'meeting-789',
  status: 'ready',
  isConnected: false,
  localParticipant: null,
  participants: [],
  error: null,
};

export const mockAppointmentsList = [
  mockAppointment,
  {
    ...mockAppointment,
    _id: '507f1f77bcf86cd799439019',
    date: '2024-03-01',
    startTime: '2:00 PM',
    endTime: '2:30 PM',
    status: 'pending',
    paymentStatus: 'pending',
    isOngoing: false,
  },
  {
    ...mockAppointment,
    _id: '507f1f77bcf86cd799439020',
    date: '2024-01-15',
    startTime: '11:00 AM',
    endTime: '11:30 AM',
    status: 'completed',
    paymentStatus: 'paid',
    isOngoing: false,
  },
];

export const mockNavigationProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    reset: jest.fn(),
    setOptions: jest.fn(),
  },
  route: {
    params: {},
  },
};