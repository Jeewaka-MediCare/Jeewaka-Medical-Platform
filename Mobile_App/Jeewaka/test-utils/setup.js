import React from 'react';
import '@testing-library/jest-native/extend-expect';
import 'react-native-gesture-handler/jestSetup';

// Mock Firebase Auth - used by components for authentication
jest.mock('firebase/auth', () => ({
  getAuth: jest.fn(() => ({
    currentUser: null,
    onAuthStateChanged: jest.fn(),
  })),
  getReactNativePersistence: jest.fn(() => 'mockPersistence'),
  initializeAuth: jest.fn(() => ({
    currentUser: null,
    onAuthStateChanged: jest.fn(),
  })),
  onAuthStateChanged: jest.fn(),
  signInWithEmailAndPassword: jest.fn(),
  createUserWithEmailAndPassword: jest.fn(),
  signOut: jest.fn(),
}));

// Mock Firebase App
jest.mock('firebase/app', () => ({
  initializeApp: jest.fn(() => ({})),
  getApps: jest.fn(() => []),
  getApp: jest.fn(() => ({})),
}));

// Mock react-native-gifted-charts - used by components for charts
jest.mock('react-native-gifted-charts', () => ({
  LineChart: jest.fn(({ children, ...props }) => null),
  BarChart: jest.fn(({ children, ...props }) => null),
  PieChart: jest.fn(({ children, ...props }) => null),
}));

// Mock Expo modules
jest.mock('expo-router', () => ({
  useRouter: () => ({
    push: jest.fn(),
    replace: jest.fn(),
    back: jest.fn(),
    canGoBack: jest.fn(() => true),
  }),
  useLocalSearchParams: () => ({}),
  router: {
    push: jest.fn(),
    replace: jest.fn(),
    back: jest.fn(),
  },
  Stack: {
    Screen: ({ children, ...props }) => children || null,
  },
  Tabs: {
    Screen: ({ children, ...props }) => children || null,
  },
}));

jest.mock('expo-font', () => ({
  useFonts: () => [true],
}));

jest.mock('expo-status-bar', () => ({
  StatusBar: 'StatusBar',
}));

jest.mock('expo-splash-screen', () => ({
  hideAsync: jest.fn(),
  preventAutoHideAsync: jest.fn(),
}));

jest.mock('expo-constants', () => ({
  default: {
    expoConfig: {
      extra: {},
    },
  },
}));

// Mock AsyncStorage
jest.mock('@react-native-async-storage/async-storage', () => ({
  getItem: jest.fn(() => Promise.resolve(null)),
  setItem: jest.fn(() => Promise.resolve()),
  removeItem: jest.fn(() => Promise.resolve()),
  clear: jest.fn(() => Promise.resolve()),
}));

// Mock Reanimated
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => {};
  return Reanimated;
});

// Mock Safe Area Context
jest.mock('react-native-safe-area-context', () => ({
  SafeAreaProvider: ({ children }) => children,
  SafeAreaView: ({ children }) => children,
  useSafeAreaInsets: () => ({ top: 0, right: 0, bottom: 0, left: 0 }),
}));

// Mock Vector Icons - this handles all icon families used by components
jest.mock('@expo/vector-icons', () => {
  const React = require('react');
  const { Text } = require('react-native');
  
  const createMockIcon = (iconFamily) => {
    const MockIcon = (props) => 
      React.createElement(Text, { testID: `${iconFamily}-${props.name || 'icon'}` }, props.name || 'Icon');
    MockIcon.displayName = iconFamily;
    return MockIcon;
  };

  return {
    Ionicons: createMockIcon('Ionicons'),
    MaterialCommunityIcons: createMockIcon('MaterialCommunityIcons'),
    MaterialIcons: createMockIcon('MaterialIcons'),
    Feather: createMockIcon('Feather'),
    FontAwesome: createMockIcon('FontAwesome'),
    FontAwesome5: createMockIcon('FontAwesome5'),
    AntDesign: createMockIcon('AntDesign'),
    Entypo: createMockIcon('Entypo'),
    EvilIcons: createMockIcon('EvilIcons'),
    Foundation: createMockIcon('Foundation'),
    Octicons: createMockIcon('Octicons'),
    SimpleLineIcons: createMockIcon('SimpleLineIcons'),
    Zocial: createMockIcon('Zocial'),
  };
});

// Silence the warning about act() not being wrapped
const originalConsoleError = console.error;
beforeAll(() => {
  console.error = (...args) => {
    if (
      typeof args[0] === 'string' &&
      args[0].includes('Warning: ReactDOM.render is no longer supported')
    ) {
      return;
    }
    originalConsoleError.call(console, ...args);
  };
});

afterAll(() => {
  console.error = originalConsoleError;
});

// Global test timeout
jest.setTimeout(10000);