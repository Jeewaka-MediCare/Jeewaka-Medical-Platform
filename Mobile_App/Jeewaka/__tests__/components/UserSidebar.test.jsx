import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { renderWithProviders } from '../../test-utils/customRender';
import UserSidebar from '../../components/UserSidebar';
import useAuthStore from '../../store/authStore';

// Mock the auth store
jest.mock('../../store/authStore');

const mockUser = {
  name: 'John Doe',
  email: 'john@example.com',
};

const mockOnLogin = jest.fn();
const mockOnLogout = jest.fn();
const mockOnClose = jest.fn();

describe('UserSidebar', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('when user is logged in', () => {
    beforeEach(() => {
      useAuthStore.mockReturnValue({
        user: mockUser,
        userRole: 'patient',
      });
    });

    it('displays user information', () => {
      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      expect(getByText(mockUser.name)).toBeTruthy();
      expect(getByText('Patient')).toBeTruthy();
    });

    it('displays avatar with user initial', () => {
      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      expect(getByText('J')).toBeTruthy(); // First letter of John
    });

    it('calls onLogout when logout button is pressed', () => {
      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      const logoutButton = getByText('Logout');
      fireEvent.press(logoutButton);
      
      expect(mockOnLogout).toHaveBeenCalledTimes(1);
    });

    it('shows logout button', () => {
      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      expect(getByText('Logout')).toBeTruthy();
    });
  });

  describe('when user is not logged in', () => {
    beforeEach(() => {
      useAuthStore.mockReturnValue({
        user: null,
        userRole: null,
      });
    });

    it('displays guest message', () => {
      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      expect(getByText('Not logged in')).toBeTruthy();
    });

    it('shows login button', () => {
      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      expect(getByText('Login')).toBeTruthy();
    });

    it('calls onLogin when login button is pressed', () => {
      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      const loginButton = getByText('Login');
      fireEvent.press(loginButton);
      
      expect(mockOnLogin).toHaveBeenCalledTimes(1);
    });
  });

  describe('role display', () => {
    it('displays Doctor role correctly', () => {
      useAuthStore.mockReturnValue({
        user: mockUser,
        userRole: 'doctor',
      });

      const { getByText } = renderWithProviders(
        <UserSidebar 
          visible={true}
          onClose={mockOnClose}
          onLogin={mockOnLogin}
          onLogout={mockOnLogout}
        />
      );
      
      expect(getByText('Doctor')).toBeTruthy();
    });
  });

  it('handles visibility prop correctly', () => {
    // Set up mock for this test
    useAuthStore.mockReturnValue({
      user: mockUser,
      userRole: 'patient',
    });

    const { rerender } = renderWithProviders(
      <UserSidebar 
        visible={false}
        onClose={mockOnClose}
        onLogin={mockOnLogin}
        onLogout={mockOnLogout}
      />
    );

    // When visible is false, the modal should not be visible
    // This is more of a structural test since Modal visibility is hard to test
    
    rerender(
      <UserSidebar 
        visible={true}
        onClose={mockOnClose}
        onLogin={mockOnLogin}
        onLogout={mockOnLogout}
      />
    );

    // Modal should now be visible
    expect(mockOnClose).not.toHaveBeenCalled();
  });

  it('matches snapshot when user is logged in', () => {
    useAuthStore.mockReturnValue({
      user: mockUser,
      userRole: 'patient',
    });

    const tree = renderWithProviders(
      <UserSidebar 
        visible={true}
        onClose={mockOnClose}
        onLogin={mockOnLogin}
        onLogout={mockOnLogout}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when user is not logged in', () => {
    useAuthStore.mockReturnValue({
      user: null,
      userRole: null,
    });

    const tree = renderWithProviders(
      <UserSidebar 
        visible={true}
        onClose={mockOnClose}
        onLogin={mockOnLogin}
        onLogout={mockOnLogout}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});