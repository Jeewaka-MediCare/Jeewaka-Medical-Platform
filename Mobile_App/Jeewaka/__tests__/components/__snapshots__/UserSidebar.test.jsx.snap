// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSidebar matches snapshot when user is logged in 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingRight": 16,
        "paddingTop": 60,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 12,
          "elevation": 8,
          "minWidth": 200,
          "opacity": 0,
          "padding": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 8,
          "transform": [
            {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#008080",
              "borderRadius": 18,
              "height": 36,
              "justifyContent": "center",
              "marginRight": 12,
              "width": 36,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            J
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1F2937",
                "fontSize": 16,
                "fontWeight": "600",
                "marginBottom": 2,
              }
            }
          >
            John Doe
          </Text>
          <Text
            style={
              {
                "color": "#64748B",
                "fontSize": 12,
                "textTransform": "capitalize",
              }
            }
          >
            Patient
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#E2E8F0",
            "height": 1,
            "marginVertical": 12,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          testID="Ionicons-log-out-outline"
        >
          log-out-outline
        </Text>
        <Text
          style={
            {
              "color": "#EF4444",
              "fontSize": 14,
              "fontWeight": "500",
              "marginLeft": 8,
            }
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`UserSidebar matches snapshot when user is not logged in 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingRight": 16,
        "paddingTop": 60,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 12,
          "elevation": 8,
          "minWidth": 200,
          "opacity": 0,
          "padding": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 8,
          "transform": [
            {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          }
        }
      >
        <Text
          testID="Ionicons-person-circle-outline"
        >
          person-circle-outline
        </Text>
        <Text
          style={
            {
              "color": "#64748B",
              "fontSize": 14,
              "marginLeft": 8,
            }
          }
        >
          Not logged in
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#E2E8F0",
            "height": 1,
            "marginVertical": 12,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          testID="Ionicons-log-in-outline"
        >
          log-in-outline
        </Text>
        <Text
          style={
            {
              "color": "#008080",
              "fontSize": 14,
              "fontWeight": "500",
              "marginLeft": 8,
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
